### PR TITLE
Add explanation of router modes

### LINKED ISSUE #445: Add a section explaining history mode

### PR DESCRIPTION
**Pull request by iWader**
Added an explanation of the router modes.

Resolves #445

### CODE REVIEW

**Comment by posva** [comment]
That's cool! Thanks @iWader

**Comment by posva** [comment]
I'm not sure if that resolves #445 thought. I think the issue is about adding more about how to setup history mode properly instead or in addition to linking to a blog post.

File: docs/en/options.md [hunk 1]
```diff
@@ -2,28 +2,19 @@
...
 
-#### history
-
-- default: false
-
-  Enables HTML5 history mode. Leverages `history.pushState()` and `history.replaceState()` for history management.
-
-  **Note**: when using the history mode, the server needs to be [properly configured](http://readystate4.com/2012/05/17/nginx-and-apache-rewrite-to-support-html5-pushstate/) so that a user directly visiting a deep link on your site doesn't get a 404.
-
-  **Note**: if you plan to serve your files with the `file://` protocol (for example, when using with [Electron](electron.atom.io)), enabling this mode will break your app. That is because your app URLs would look like `file:///path/to/your/app/index.html/my-route/`, which makes no sense in UNIX filesystem architecture.
```
Comment (fnlctrl) [inline docs/en/options.md:20]: Why is this note removed?

**Comment by fnlctrl** [comment]
Sorry, but in my opinion this PR didn't actually solve anything... other than changing some titles to fit 2.0.

**Comment by posva** [comment]
@fnlctrl Sorry, I should have let you review it too. I'll ping you next time before merging.
The file is not used yet in the new docs. I agree, those notes have to be kept
I merged a too fast. Let me add back the notes

**Comment by fnlctrl** [comment]
@posva No problem man.

**Comment by iWader** [comment]
I'll happily submit another PR tonight, I wasn't aware that's what the
issue was asking for. I though it was just to update on the new options

**Comment by posva** [comment]
@iWader About the notes, it's already fixed, don't worry 😉

**Comment by iWader** [comment]
@posva I mean't about the further info on setting up the server for the history mode.

Should I put this into a separate file and link to it in place of the blog post? This would allow for expansion in the future for further webservers, etc.

Are there any specific servers to include, or just nginx and apache?

**Comment by posva** [comment]
I really don't know what was the purpose of the issue in the first place, so I cannot help you further. IMO notes about configuring common servers configuration (express, nginx and apache) is what may help. It should go in a different section too.
However, the issue #445 may be about explaining how history mode works and how to integrate with HTML5 history. Let me post a message on that issue.

**Pull request merged**